### DISCUSSION TITLE
Lua equivalent to `%{echo:...}` and similar

### DISCUSSION THREAD

**Discussion by voxik** [rpm-software-management/rpm#2968] [Q&A] [answered]
Currently the way to get some information from Lua macro is quite clumsy. One needs to do something like:

~~~
%define foo() %{lua:
  rpm.expand("%{echo: some output}")
}
~~~

Would it be possible to make it easier, e.g. something like this would be more convenient:
~~~
%define foo() %{lua:
  rpm.echo("some output")
}
~~~

Or maybe just:

~~~
%define foo() %{lua:
  echo("some output")
}
~~~

**Comment by mlschroe** [comment, marked as answer]
You can already do:
```
macros.echo({"hello world"})
```

**Comment by voxik** [comment]
Oh. Interesting, the for the tip 👍 Nevertheless, I was not able to figure that out myself, therefore it would deserve to be documented if that is the right approach.

**Comment by voxik** [comment]
TBH figuring out `rpm.expand("%{echo: some output}")` is also not that straight forward. I was able to find such example at least.

**Comment by pmatilai** [comment]
Accessing rpm macros through the `macros` table *is* documented in https://rpm-software-management.github.io/rpm/manual/lua.html

**Reply by voxik** [reply]
Yes, I don't argue with that (and I was aware about that). But output from Lua to terminal is useful and would IMHO deserve some separate note.
